Automattic/jetpack
# My Jetpack: Redirect to post_activation_url after activation

Implements the redirect to the `post_activation_url` defined by each product.

This redirect takes effect when the user activates the product from the interstitial page.

This also fix a bug in the Backup purchase flow. The Backup purchase flow was not working properly when the user was not connected to WPCOM.

With this PR, if the user is not connected, they will be taken to the plugin's page instead of the checkout.

#### Changes proposed in this Pull Request:

* Redirect to post_activation_url after product activation from iterstitial screen

#### Jetpack product discussion


N/A

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* On a fresh site, we have to test adding each product from the interstitial page. 
* You might need to cancel your subscription, or delete the plugin to be able to do this for every product, as some products share the same plugin 

* Backup
 * Without a user connection, you should be redirected to the plugin's page (if Backup plugin is there) or to the Jetpack Dashboard (if Backup plugin is not there
 * With a user connection, you should be redirected to the checkout page to buy Backup

* Search -> you should be redirected to the checkout page to buy Search

* Scan -> you should be redirected to the checkout page to buy Scan

* VideoPress -> You should stay in My Jetpack page

* Anti-Spam -> You should go to Akismet page

* CRM -> You should go to CRM Welcome page

* Boost -> You should go to Boost page
